gcp_variant_transforms/libs/vcf_header_parser.py

```diff
@@ -32,68 +32,29 @@
-  """Returns merged VCF headers (FORMAT and INFO) from ``input_pattern``.
-
-  If multiple files are specified by ``input_pattern`` then header fields will
-  be merged by key and only one will be chosen as representative (in no
-  particular order). If the same key is used in multiple files, then all types
-  and numbers must be the same.
+def get_vcf_headers(input_file):
+  """Returns VCF headers (FORMAT and INFO) from ``input_file``.
 
   Args:
-    input_pattern (str): A string specifying the path to VCF input file(s). They
-      can be local or remote (e.g. on GCS).
+    input_file (str): A string specifying the path to a representative VCF
+    file. It can be local or remote (e.g. on GCS).
   Returns:
-    `HeaderFields`` specifying merged header info from all files in
-    ``input_pattern``.
-  Raises:
-    ValueError: If the VCF headers are invalid or incompatible (e.g. the same
-      key if defined with different types in multiple files).
-  """
-  merged_info_fields = {}
-  merged_format_fields = {}
-  # The match implementation supports one pattern, but the API technically
-  # supports multiple patterns, hence the need for a list (in both request
-  # and response).
-  match_results = FileSystems.match([input_pattern])
-  if not match_results:
-    return HeaderFields(merged_info_fields, merged_format_fields)
-  for file_metadata in match_results[0].metadata_list:
-    file_name = file_metadata.path
-    if not file_metadata.size_in_bytes:
-      continue  # Ignore empty files.
-    try:
-      vcf_reader = vcf.Reader(fsock=_line_generator(file_name))
-    except SyntaxError as e:
-      raise ValueError('Invalid VCF header: %s' % str(e))
-    _merge_header_fields(merged_info_fields, vcf_reader.infos)
-    _merge_header_fields(merged_format_fields, vcf_reader.formats)
-  return HeaderFields(merged_info_fields, merged_format_fields)
-
-
-def _merge_header_fields(source, to_merge):
-  """Modifies ``source`` to add any keys from ``to_merge`` not in ``source``.
-
-  Args:
-    source (dict): Source header fields.
-    to_merge (dict): Header fields to merge with ``source``.
+    ``HeaderFields`` specifying header info. Empty is returned if
+    ``input_file`` doesn't exist or is empty.
   Raises:
-    ValueError: If the header fields are incompatible (e.g. same key with
-      different types or numbers).
+    ValueError: If ``input_file`` is not a valid VCF file.
   """
-  if not to_merge:
-    return source
-  for key, to_merge_value in to_merge.iteritems():
-    if key not in source:
-      source[key] = to_merge_value
-    else:
-      source_value = source[key]
-      if (source_value.num != to_merge_value.num or
-          source_value.type != to_merge_value.type):
-        raise ValueError(
-            'Incompatible number of types in header fields: %s, %s' % (
-                source_value, to_merge_value))
-
+  # Return empty if file doesn't exist.
+  if not FileSystems.exists(input_file):
+    return HeaderFields({}, {})
```

Comment:
It seems better to raise an error in this case because it either means that the merge_headers pipeline failed or the user input the path to their representative header file incorrectly in which case the job is likely to fail later in the pipeline.

Reply:
Agreed. The current implementation returns empty in this case, and I was not sure if this is intended or not, otherwise I would raise an error.  Done!